pkg/deploy/lattice/target_group_manager.go

```diff
@@ -247,70 +248,102 @@ type tgListOutput struct {
 func (s *defaultTargetGroupManager) List(ctx context.Context) ([]tgListOutput, error) {
 	lattice := s.cloud.Lattice()
 	var tgList []tgListOutput
-	targetGroupListInput := vpclattice.ListTargetGroupsInput{}
+	targetGroupListInput := vpclattice.ListTargetGroupsInput{
+		VpcIdentifier:   aws.String(config.VpcID),
+		TargetGroupType: aws.String(vpclattice.TargetGroupTypeIp),
+	}
 	resp, err := lattice.ListTargetGroupsAsList(ctx, &targetGroupListInput)
 	if err != nil {
 		return nil, err
 	}
-
+	if len(resp) == 0 {
+		return nil, nil
+	}
+	tgArns := utils.SliceMap(resp, func(tg *vpclattice.TargetGroupSummary) *string { return tg.Arn })
+	taggingApiResp, err := s.cloud.TaggingApi().GetResourcesWithContext(ctx, &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceARNList: tgArns,
+	})
+	tgArnToTagsMap := make(map[string][]*resourcegroupstaggingapi.Tag)
+	for _, tagMapping := range taggingApiResp.ResourceTagMappingList {
+		tgArnToTagsMap[*tagMapping.ResourceARN] = tagMapping.Tags
+	}
+	if err != nil {
+		return nil, err
+	}
 	for _, tg := range resp {
-		if aws.StringValue(tg.VpcIdentifier) == config.VpcID {
-			tagsInput := vpclattice.ListTagsForResourceInput{
-				ResourceArn: tg.Arn,
-			}
-
-			tagsOutput, err := lattice.ListTagsForResourceWithContext(ctx, &tagsInput)
-			if err != nil {
-				s.log.Infof("Failed ListTags %s: %s", aws.StringValue(tg.Arn), err)
-				// setting it to nil, so the caller knows this failed
-				tagsOutput = nil
-			}
-			tgOutput := tgListOutput{
-				tgSummary:       tg,
-				targetGroupTags: tagsOutput,
-			}
-			tgList = append(tgList, tgOutput)
-		}
+		tgList = append(tgList, tgListOutput{
+			tgSummary:       tg,
+			targetGroupTags: &vpclattice.ListTagsForResourceOutput{
+				Tags: tagsMapFromTaggingApiTags(tgArnToTagsMap[*tg.Arn]),
+			},
+		})
 	}
 	return tgList, err
 }
 
+func tagsMapFromTaggingApiTags(tags []*resourcegroupstaggingapi.Tag) map[string]*string {
+	out := make(map[string]*string)
+	for _, tag := range tags {
+		out[*tag.Key] = tag.Value
+	}
+	return out
+}
+
 func (s *defaultTargetGroupManager) findTargetGroup(
 	ctx context.Context,
 	modelTargetGroup *model.TargetGroup,
-) (*vpclattice.TargetGroupSummary, error) {
-	listInput := vpclattice.ListTargetGroupsInput{
-		VpcIdentifier: aws.String(modelTargetGroup.Spec.VpcId),
-	}
-	resp, err := s.cloud.Lattice().ListTargetGroupsAsList(ctx, &listInput)
+) (*vpclattice.GetTargetGroupOutput, error) {
+	resp, err := s.cloud.TaggingApi().GetResourcesWithContext(ctx, &resourcegroupstaggingapi.GetResourcesInput{
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
```

Comment:
We have similar logic that takes tags and creates model fields, we should add the reverse in ```targetgroup.go```